src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/FindAllReferencesHandler.cs

```diff
@@ -78,35 +94,65 @@ public async Task<VSReferenceItem[]> HandleRequestAsync(ReferenceParams request,
 
             cancellationToken.ThrowIfCancellationRequested();
 
-            var serverKind = LanguageServerKind.CSharp;
-            var referenceParams = new ReferenceParams()
+            // Temporary till IProgress serialization is fixed
+            var token = Guid.NewGuid().ToString(); // request.PartialResultToken.Id
+
+            var referenceParams = new SerializableReferenceParams()
             {
                 Position = projectionResult.Position,
                 TextDocument = new TextDocumentIdentifier()
                 {
                     Uri = projectionResult.Uri
                 },
                 Context = request.Context,
-                PartialResultToken = request.PartialResultToken
+                PartialResultToken = token // request.PartialResultToken
             };
 
-            var result = await _requestInvoker.ReinvokeRequestOnServerAsync<ReferenceParams, VSReferenceItem[]>(
+            if (!_lspProgressListener.TryListenForProgress(
+                token,
+                onProgressNotifyAsync: (value) => ProcessReferenceItemsAsync(value, request.PartialResultToken, cancellationToken),
```

Comment:
Talked offline. I'm of the opinion this should probably take the cancellation token in:
```suggestion
                onProgressNotifyAsync: (value, token) => ProcessReferenceItemsAsync(value, request.PartialResultToken, token),
```

But then again, if you feel differently I'm willing to give it a shot! 😄

Reply:
I agree, would be beneficial, especially as it is a `onProgressNotify**Async**` handler, as you pointed out.